### PR TITLE
Create unit test for domain code

### PR DESCRIPTION
**Pull request by gavinkalika**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | NO
| Deprecations? | NO
| Fixed ticket? | [Issue](https://github.com/PrestaShop/PrestaShop/issues), please write "Fixes #[issue number]" here.
| How to test?  | Please indicate how to best verify that this PR is correct.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16028)

### CODE REVIEW

**Comment by gavinkalika** [comment]
related to https://github.com/PrestaShop/PrestaShop/pull/15010/

related to https://github.com/PrestaShop/PrestaShop/issues/15145

File: tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public function testItCreatesProfileWithValidValues()
+    {
+        $profileId = new ProfileId(1);
+
+        $this->assertEquals(1, $profileId->getValue());
+    }
+
+    /**
+     * @dataProvider testItExceptionThrownWithInvalidValuesData
+     ** */
```
Comment (PierreRambaud) [inline tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php]:
```suggestion
     */
```

File: tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public function testItExceptionThrownWithInvalidValues($profileId)
+    {
+        $this->expectException(ProfileException::class);
+        new ProfileId($profileId);
+    }
+
+    public function testItExceptionThrownWithInvalidValuesData()
+    {
+        return [
+            'stringy -1' => ['-1'],
```
Comment (PierreRambaud) [inline tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php]: No need to add `stringy`
```suggestion
            ['-1'],
```

File: tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    public function testItExceptionThrownWithInvalidValues($profileId)
+    {
+        $this->expectException(ProfileException::class);
+        new ProfileId($profileId);
+    }
+
+    public function testItExceptionThrownWithInvalidValuesData()
+    {
+        return [
+            '-1' => ['-1'],
```
Comment (PierreRambaud) [inline tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php]:
```suggestion
            ['-1'],
```

File: tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            '1.1' => ['1.1'],
```
Comment (PierreRambaud) [inline tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php]:
```suggestion
            ['1.1'],
```

File: tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            'a' => ['a'],
```
Comment (PierreRambaud) [inline tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php]:
```suggestion
            ['a'],
```

File: tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            '+' => ['+'],
```
Comment (PierreRambaud) [inline tests/Unit/Core/Domain/Profile/ValueObject/ProfileIdTest.php]:
```suggestion
            ['+'],
```

**Comment by PierreRambaud** [comment]
You don't need the `key => value` assignment in your data provider ;)

**Review by PierreRambaud — APPROVED** [review]

**Comment by gavinkalika** [comment]
is there a way to squash commits? I did a merge on on my pull request.... trying to update my local copy.

**Comment by PierreRambaud** [comment]
@gavinkalika `git brease -i develop` to squash commits ;) 
Of course after update your local develop branch with latest changes

**Pull request closed without merging**